httpcache.go

```diff
@@ -227,9 +229,25 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 				resp.Header.Set(fakeHeader, reqValue)
 			}
 		}
-		respBytes, err := httputil.DumpResponse(resp, true)
-		if err == nil {
-			t.Cache.Set(cacheKey, respBytes)
+		switch req.Method {
+		case "GET":
+			// Delay caching until EOF is read.
+			resp.Body = &cachedReadCloser{
+				R: resp.Body,
+				OnEOF: func(r io.ReadCloser) {
+					resp := *resp
+					resp.Body = r
```

Comment:
Then `r` would be `io.Reader` here, and we can do `resp.Body = ioutil.NopCloser(r)`.